river/metrics/base_internal_clustering.py

```diff
@@ -0,0 +1,82 @@
+import abc
+import numbers
+import typing
+
+from river import stats
+
+__all__ = ["InternalClusteringMetrics"]
+
+
+class InternalClusteringMetrics(abc.ABC):
+    """
+    Mother class of all internal clustering metrics.
+    """
+
+    # Define the format specification used for string representation.
+    _fmt = ",.6f"  # Use commas to separate big numbers and show 6 decimals
+
+    @abc.abstractmethod
+    def update(
+        self, centers, point, y_pred, sample_weight
```

Comment:
As I mentioned in the meeting yesterday, I do prefer using `.update(self, x, labels, sample_weight, centers)` instead. I believe that by using `labels` rather than `y_pred` we explicitly show that clustering is a different monster other than classification/regression. Here we are not dealing with predictive tasks, but descriptive ones.

Reply:
I also prefer `x` instead of `point` for consistency.

Reply:
Given that we are using `x` and `y` in external metrics, I suggest we keep that convention:

```
update(self, x, y_pred, sample_weight, centers)
```

For example:
```
accuracy.update(y_true, y_pred, sample_weight)
ssq.update(x, y_pred, sample_weight, centers)
```